anakondo.el

```diff
@@ -197,17 +202,28 @@ and the completion candidates for it at cdr for buffer.")
 
 We try to find the project root by:
 1. Trying to query `clojure-mode' for it.
-2. Trying to query projectile for it.
-3. Defaulting to the `default-directory' of the buffer otherwise.
+2. Trying to query `project' for it.
+3. Trying to query projectile for it.
+4. Defaulting to the `default-directory' of the buffer otherwise.
 
 Anaphoric macro, binds `root' implicitly."
-  `(let* ((root (or (clojure-project-dir)
-                    (projectile-project-root)
+  `(let* ((root (or (and (featurep 'clojure-mode) (clojure-project-dir))
+                    (anakondo--project-get-project-root)
```

Comment:
Since I'm not very familiar with `project.el`, I wonder, since it is looked up before projectile, if someone was currently relying on projectile, could project.el surprisingly take over? Like can project.el find a project root for any buffer without explicitly configuring it? If so, it might be best to add it after projectile.

Reply:
I guess yes, project.el searches for Git repositlry root, so it may find project before projectile.

I've put it before projectile, because it is inbuilt mode, and therefore will be available for most users OOTB. Another option would be to provide a custom variable which will hold a list of functions used to get project root, but this would require some logic to check if every function from the list is available to use.

I'll see what can be done here

Reply:
I have some use cases where git is the wrong project root, with projectile you can override that so a .projectile file has precedence on git repo, for when the git repo contains multiple projects.

So I thought here it would find the wrong root.

My thought is that if someone has projectile, they probably do so cause they use it over the default project.el. So it seems like it would make sense to prioritize projectile over project.el ?

Reply:
Yeah, but I also had problems with projectile returning incorrect project root, which I've described here https://github.com/bbatsov/projectile/issues/1130#issuecomment-492515263. Project.el was easy enough to configure to include search for custom project root markers, so I've never looked back on using projectile :)

Which means that neither of these projects is the source of truth regarding getting project root.

I'll reorder things as you suggest, as I think projectile is far more popular than project.el, so that's that. But I also think that perhaps Anakondo could provide a menu in Custom where user could just select what algorythm to use to find project root:

- auto - try clojure-mode, projectile, project.el, default-dir
- clojure-mode - only clojure-mode
- projectile - only projectile
- project.el - opnly project.el

What do you think?

Reply:
I think having a custom config could be nice, but I also feel we.xould wait to see.if someone does need that. Since I do find it strange to have projectile installed in Emacs if you're not using it.

Reply:
> Since I do find it strange to have projectile installed in Emacs if you're not using it.

Some package require it as a dependency